shell/platform/windows/flutter_windows_view.h

```diff
@@ -0,0 +1,245 @@
+                  double delta_x,
+                  double delta_y,
+                  int scroll_offset_multiplier);
+
+  // Set's |event_data|'s phase to either kMove or kHover depending on the
+  // current
+  // primary mouse button state.
+  void SetEventPhaseFromCursorButtonState(
+      FlutterPointerEvent* event_data) const;
+
+  // Sends a pointer event to the Flutter engine based on givern data.  Since
+  // all input messages are passed in physical pixel values, no translation is
+  // needed before passing on to engine.
+  void SendPointerEventWithData(const FlutterPointerEvent& event_data);
+
+  // Gets the current mouse state.
+  MouseState GetMouseState() const { return mouse_state_; }
+
+  // Resets the mouse state to its default values.
+  void ResetMouseState() { mouse_state_ = MouseState(); }
+
+  // Updates the mouse state to whether the last event to Flutter had at least
+  // one mouse button pressed.
+  void SetMouseFlutterStateDown(bool is_down) {
+    mouse_state_.flutter_state_is_down = is_down;
+  }
+
+  // Updates the mouse state to whether the last event to Flutter was a kAdd
+  // event.
+  void SetMouseFlutterStateAdded(bool is_added) {
+    mouse_state_.flutter_state_is_added = is_added;
+  }
+
+  // Updates the currently pressed buttons.
+  void SetMouseButtons(uint64_t buttons) { mouse_state_.buttons = buttons; }
+
+  std::unique_ptr<AngleSurfaceManager> surface_manager = nullptr;
+  EGLSurface render_surface = EGL_NO_SURFACE;
+
+  // state of the mouse button.
+  bool pointer_is_down_ = false;
+
+  // The handle to the Flutter engine instance.
+  FLUTTER_API_SYMBOL(FlutterEngine) engine_ = nullptr;
+
+  // Whether or not to track mouse movements to send kHover events.
+  bool hover_tracking_is_enabled_ = false;
+
+  // Whether or not the pointer has been added (or if tracking is enabled, has
+  // been added since it was last removed).
+  bool pointer_currently_added_ = false;
+
+  // Set to true to be notified when the mouse leaves the window.
+  bool tracking_mouse_leave_ = false;
+
+  // Keeps track of mouse state in relation to the window.
+  MouseState mouse_state_;
+
+  // The window handle given to API clients.
+  std::unique_ptr<FlutterDesktopView> window_wrapper_;
+
+  // The plugin registrar handle given to API clients.
+  std::unique_ptr<FlutterDesktopPluginRegistrar> plugin_registrar_;
+
+  // Message dispatch manager for messages from the Flutter engine.
+  std::unique_ptr<flutter::IncomingMessageDispatcher> message_dispatcher_;
+
+  // The plugin registrar managing internal plugins.
+  std::unique_ptr<flutter::PluginRegistrar> internal_plugin_registrar_;
+
+  // Handlers for keyboard events from Windows.
+  std::vector<std::unique_ptr<flutter::KeyboardHookHandler>>
+      keyboard_hook_handlers_;
+
+  // Handler for the flutter/platform channel.
+  std::unique_ptr<flutter::PlatformHandler> platform_handler_;
+
+  // should we forword input messages or not
+  bool process_events_ = false;
```

Comment:
Not sure how to determine if it's needed or not.  I can rip it out, please advise what to test to ensure we are not regressing anything.

Reply:
IIRC the case was that if we showed a modal GTK window (e.g., file chooser), clicking on the Flutter window would actually send clicks to the window even though the window wouldn't come forward, so we needed to manually suppress events. I think it was just a hack because of the GTK/GLFW mismatch.

Reply:
OK.  I will rip out process_events_ and all references to it then ;-)

Reply:
Looking in further, truly ripping this out would require updating IncomingMessageDispatcher in common.  That could require cascading changes which seem like beyond the scope of this change.  I can a) rip it out locally by updating HandleMessage with empty delegates and removing the local variable ignoring input blocking b) leave as is and address in a subsequent change which specifically removes the concept of blocking from all consumers of the common header c) do the full rip out as part of this change.

Reply:
Ah, right. Let's do (b) while I think about whether we still want it for the GLFW embedding.

Reply:
Fixed in next commit